pkg/query-service/agentConf/db.go

```diff
@@ -0,0 +1,227 @@
+		id, 
+		version, 
+		element_type, 
+		COALESCE(created_by, -1) as created_by, 
+		active, 
+		is_valid, 
+		disabled, 
+		deploy_status, 
+		deploy_result 
+		FROM agent_config_versions 
+		WHERE element_type = $1`, typ)
+
+	return c, err
+}
+
+func (r *Repo) GetConfigVersion(ctx context.Context, typ ElementTypeDef, v int) (*ConfigVersion, error) {
+	var c ConfigVersion
+	err := r.db.GetContext(ctx, &c, `SELECT 
+		id, 
+		version, 
+		element_type, 
+		COALESCE(created_by, -1) as created_by, 
+		COALESCE((SELECT NAME FROM users 
+		WHERE id = v.created_by), "unknown") created_by_name,
+		active, 
+		is_valid, 
+		disabled, 
+		deploy_status, 
+		deploy_result,
+		last_hash,
+		last_config
+		FROM agent_config_versions v 
+		WHERE element_type = $1 
+		AND version = $2`, typ, v)
+
+	return &c, err
+
+}
+
+func (r *Repo) GetLatestVersion(ctx context.Context, typ ElementTypeDef) (*ConfigVersion, error) {
+	var c ConfigVersion
+	err := r.db.GetContext(ctx, &c, `SELECT 
+		id, 
+		version, 
+		element_type, 
+		COALESCE(created_by, -1) as created_by, 
+		active, 
+		is_valid, 
+		disabled, 
+		deploy_status, 
+		deploy_result 
+		FROM agent_config_versions 
+		WHERE element_type = $1 
+		AND version = ( 
+			SELECT MAX(version) 
+			FROM agent_config_versions 
+			WHERE element_type=$2)`, typ, typ)
+	if err != nil {
+		zap.S().Error("failed get latest config version for element:", typ, err)
+	}
+	return &c, err
+}
+
+func (r *Repo) insertConfig(ctx context.Context, c *ConfigVersion, elements []string) (fnerr error) {
+
+	if string(c.ElementType) == "" {
+		return fmt.Errorf("element type is required for creating agent config version")
+	}
+
+	if len(elements) == 0 {
+		zap.S().Error("insert config called with no elements", c.ElementType)
+		return fmt.Errorf("config must have atleast one element")
+	}
+
+	if c.Version != 0 {
+		// the version can not be set by the user, we want to auto-assign the versions
+		// in a monotonically increasing order starting with 1. hence, we reject insert
+		// requests with version anything other than 0. here, 0 indicates un-assigned
+		zap.S().Error("invalid version assignment while inserting agent config", c.Version, c.ElementType)
+		return fmt.Errorf("user defined versions are not supported in the agent config")
+	}
+
+	configVersion, err := r.GetLatestVersion(ctx, c.ElementType)
+	if err != nil {
+		if err != sql.ErrNoRows {
+			zap.S().Error("failed to fetch latest config version", err)
+			return fmt.Errorf("failed to fetch latest config version")
+		}
+	}
+
+	c.Version = updateVersion(configVersion.Version)
+
+	defer func() {
+		if fnerr != nil {
+			// remove all the damage (invalid rows from db)
+			r.db.Exec("DELETE FROM agent_config_versions WHERE id = $1", c.ID)
+			r.db.Exec("DELETE FROM agent_config_elements WHERE version_id=$1", c.ID)
+		}
+	}()
+
+	// insert config
+	configQuery := `INSERT INTO agent_config_versions(	
+		id, 
+		version, 
+		element_type, 
```

Comment:
The created_by value is not inserted here, can you please checkout my changes from here https://github.com/SigNoz/signoz/blob/94333d6152a09b8c2809ff735e00f532b973d7f5/pkg/query-service/agentConf/db.go

Reply:
Added created_by